src/main/java/session/FormSession.java

```diff
@@ -429,4 +418,23 @@ public void setAsUser(String asUser) {
     public boolean getOneQuestionPerScreen() {
         return oneQuestionPerScreen;
     }
+
+    public void reload(FormDef formDef, String postUrl) throws IOException {
+        if(getInstanceXml() != null){
+            loadInstanceXml(formDef, getInstanceXml());
+            initialize(false, sessionData);
+        } else {
+            initialize(true, sessionData);
+        }
+        if (this.oneQuestionPerScreen) {
```

Comment:
its not right now - this is used by session replaying in that other PR. I'm gonna leave fixing/cleaning this up to that one